core/io/image_loader.cpp

```diff
@@ -62,7 +62,7 @@ Error ImageLoader::load_image(String p_file, Ref<Image> p_image, FileAccess *p_c
 
 		if (!loader[i]->recognize(extension))
 			continue;
-		Error err = loader[i]->load_image(p_image, f, p_force_linear);
+		Error err = loader[i]->load_image(p_image, f, p_force_linear, p_scale);
```

Comment:
adding this p_scale to each function might be confusing. because in most load_image functions it does not do anything. it's just there so svgs can use it.

What about adding a function called load_image_scaled(String p_file, Ref<Image> p_image, FileAccess *p_custom, bool p_force_linear, float p_scale)
which than called load_image and resizes right after the image got loaded and returns the resized version **OR** when the extension is ".svg" it know it can also call the load_image_scaled function.

another solution:
Keep the p_scale in ImageLoader.
Than check if the typeid(loader[I]) == typeid(ImageLoaderSVG)  and if this is the case cast it to ImageLoaderSVG and call load_image('all params plus the scale param') which only exists in the ImageLoaderSVG.
if that is not the case:
{
   load_image('all params without scale');
   p_image.resize(p_scale);
}